vault/expiration.go

```diff
@@ -1815,7 +1854,16 @@ func (m *ExpirationManager) loadEntry(ctx context.Context, leaseID string) (*lea
 	} else {
 		ctx = namespace.ContextWithNamespace(ctx, namespace.RootNamespace)
 	}
-	return m.loadEntryInternal(ctx, leaseID, restoreMode, true)
+
+	//
+	// If a lease entry is nil, proactively delete the lease lock, in case we
+	// created one erroneously.
+	leaseEntry, err := m.loadEntryInternal(ctx, leaseID, restoreMode, true)
+	if err != nil && leaseEntry == nil {
+		m.deleteLockForLease(leaseID)
```

Comment:
why wouldn't we also want to do this if error is nil?

Reply:
Hm.... this is backwards, actually, thanks for asking about it. 

If err != nil, then we don't know whether the entry is there or not, so we should not assume that it isn't.